Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/server/SurveyServer.kt

```diff
@@ -0,0 +1,36 @@
+package de.rki.coronawarnapp.datadonation.survey.server
+
+import com.google.protobuf.ByteString
+import dagger.Lazy
+import de.rki.coronawarnapp.datadonation.OneTimePassword
+import de.rki.coronawarnapp.server.protocols.internal.ppdd.EdusOtp
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class SurveyServer @Inject constructor(
+    private val surveyApi: Lazy<SurveyApiV1>,
+    private val dispatcherProvider: DispatcherProvider
+) {
+
+    private val api: SurveyApiV1
+        get() = surveyApi.get()
+
+    suspend fun authOTP(
+        data: OneTimePassword
+    ): SurveyApiV1.DataDonationResponse = withContext(dispatcherProvider.IO) {
+        Timber.d("authOTP()")
+
+        val dataDonationPayload = EdusOtp.EDUSOneTimePassword.newBuilder()
```

Comment:
I think we need to send `EdusOtpRequestAndroid.EDUSOneTimePasswordRequestAndroid` here? This one also includes the SafetyNet Result which then also needs to be passed to `authOTP()`

Reply:
thanks!